### PR TITLE
added tasks.json and bash scripts for better UX improvement in vsCode

### PR DESCRIPTION
**Pull request by eneag-sf**
This pull requests contains the scripts/files needed to make dx@scale & sfdx more scalable towards new learners, and onboard new learners easier.

There is no need anymore to open the terminal and write commands to fetch scratch orgs.

Screenshot 2023-01-23 at 11 57 10

Prerequisites: 
Open terminal and give access to the bash comments so you do not receive a "_Permission denied_" error.

Here the commands to write in terminal (only once needed):

$ chmod +x ./scripts/controllerScratchOrg.sh    
$ chmod +x ./scripts/fetchScratchOrg.sh

Important: Your source control should be empty otherwise you can't checkout to a new brach 🔺

How to use the fetch scratch org feature:

1. Open the repo folder in vscode.
2. Authenticate in DevHub
3. Click on Terminal > Run Tasks > select "Fetch new Scratch Org 🚀"
4. Enter your initials and press Enter (example: EG)
5. Enter your JIRA ticket number (example: 101)
6. Terminal will open automatically and you will be checked out to a new branched called **feature/EG-101**

How to use the fetch scratch org feature:

1. Develop in your scratch org
2. Click on Terminal > Run Tasks > select "Pull your changes from the Scratch Org ⬅️"
3. Terminal will open automatically and will pull your changes into your local branch.
4. Make adjustments, stage, commit and open PRs.

Enjoy 🎉

### CODE REVIEW

**Comment by eneag-sf** [comment]
As this feature branch is not yet merged to the main branch, the script controllerScratchOrg.sh has a command that checkouts to the main branch, pulls from the main origin and then checkouts to the feature branch to ensure that the feature branch is up-to-date with the main branch.

After the PR is tested, the lines from 24-28 can be removed and the lines 18-22 should be the prod ready commands.

**Comment by eneag-sf** [comment]
Also please make sure to adjust the commands. For example, in the script, I have: 

sfdx sfpowerkit:pool:fetch -t **dev**

Maybe you have another tag for the dev branch, so feel free to adjust the commands to your projects needs.

**Comment by TimPaulaskasDS** [comment]
#!/usr/bin/env sh

Using "#!/usr/bin/env sh" is generally considered safer in shell scripts when you have users that could be using different shells, such as zsh or bash. This is because /usr/bin/env is a tool that is present on most UNIX-like systems, and it allows you to run a command in an environment where the command is looked for in the PATH, regardless of which shell the user is currently running. So, if the user is running zsh or bash, the script will still execute properly.

**Comment by eneag-sf** [comment]
Thanks for the suggestion @TimPaulaskasDS. I have now updated the scripts.

File: scripts/fetchScratchOrg.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  then
+    echo "No argument for a branch name. Recommended format is <initials of developer>.<User story id>"
+    exit 1
+fi
+
+# Fetch Scratch Org from DEV Pool
+# Add authInfo.txt to the .gitignore so it is not saved in the repo just to be sure
+# The script will remove the files anyway (line 43)
+# I couldn't use the standard json file that sfpowerscipt uses for the authentication because of the timing
+sfdx sfpowerkit:pool:fetch -t dev > authInfo.txt
```
Comment (azlam-abdulsalam) [inline scripts/fetchScratchOrg.sh:13]: sfpowerkit is deprecated, need to use sfpowerscripts. Are you facing any issues here?

Comment (azlam-abdulsalam) [reply scripts/fetchScratchOrg.sh:13]: sfpowerscripts along with dev pool will synchronise the working directory

Comment (eneag-sf) [reply scripts/fetchScratchOrg.sh]: you are right. I have adjusted the command and replaced it with `sfdx sfpowerscripts:pool:fetch`

**Review by azlam-abdulsalam — APPROVED** [review]
Thanks @eneag-sf

**Pull request merged**